src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CreateRoleCommand.java

```diff
@@ -41,16 +41,16 @@ public DataverseRole execute(CommandContext ctxt) throws CommandException {
         //Test to see if the role already exists in DB
         try {
             DataverseRole testRole = ctxt.em().createNamedQuery("DataverseRole.findDataverseRoleByAlias", DataverseRole.class)
-                    .setParameter("alias", created.getAlias())
+                    .setParameter("alias", role.getAlias())
                     .getSingleResult();
-            if (!(testRole == null)) {
+            if (testRole != null && !testRole.getId().equals(role.getId())) {
                 throw new IllegalCommandException(BundleUtil.getStringFromBundle("permission.role.not.created.alias.already.exists"), this);
             }
         } catch (NoResultException nre) {
             //  we want no results because that meand we can create a role
```

Comment:
small typo but I assume that mean**s** and not meand :)

Reply:
@DS-INRA good catch. Fixed in 89549d7.